runtime/lua/vim/_defaults.lua

```diff
@@ -107,22 +107,39 @@ do
           vim.inspect(cmd.cmd)
         )
       end
+      return err
+    end
 
-      if err then
-        vim.notify(err, vim.log.levels.ERROR)
+    --- Gets the URL at cursor, if any.
+    local function get_url()
+      local current_node = vim.treesitter.get_node { lang = 'markdown_inline' }
```

Comment:
You need to make sure the tree is actually parsed before doing this; see `:h vim.treesitter.get_node()`.

(A running treesitter highlighter (`vim.treesitter.start()`) will do that for you via the callbacks, but otherwise you're on your own.)

And you shouldn't force the language here; set the correct parser (which normally is `markdown`, not `markdown_inline`), otherwise all manner of bad things can happen. 

And note that URLs in Markdown have to be... marked up (`<url>`), otherwise they are not valid nodes and won't be recognized by the parsers.

In sum, I don't think tree-sitter is the right tool for the job here unless you _know_ that you are dealing with valid markdown (in a markdown file or injected).

Reply:
Let's get this working on markdown buffers if nothing else.

Reply:
Then call `parser = vim.treesitter.get_parser(bufnr)` followed by `parser:parse(range)` before `get_node()`. Make sure to only call `parse` for the minimal `range` you need (so `get_url` should get the cursor position _first_).